### PR TITLE
Implement OCI Image format puller

### PR DESCRIPTION
**Pull request by xwinxu**
Extended rules_docker API to now contain the `Write` function which outputs the images from the puller in OCI standard format. BUILD files updated in the process.

### CODE REVIEW

File: container/new_pull.bzl [hunk 1]
```diff
@@ -0,0 +1,209 @@
+# Copyright 2017 The Bazel Authors. All rights reserved.
```
Comment (xingao267) [inline container/new_pull.bzl]: I think you probably don't want the starlark part in this PR?

File: container/go/cmd/puller/puller.go [hunk 2]
```diff
@@ -12,27 +12,30 @@
...
+	"github.com/google/go-containerregistry/pkg/v1/cache"
 	"github.com/google/go-containerregistry/pkg/v1/remote"
-	"github.com/google/go-containerregistry/pkg/v1/tarball"
 )
 
 var (
 	imgName         = flag.String("name", "", "The name location including repo and digest/tag of the docker image to pull and save. Supports fully-qualified tag or digest references.")
 	directory       = flag.String("directory", "", "Where to save the images files.")
 	clientConfigDir = flag.String("client-config-dir", "", "The path to the directory where the client configuration files are located. Overiddes the value from DOCKER_CONFIG.")
+	cachePath       = flag.String("cache", "", "Image's files cache directory.")
```
Comment (xingao267) [inline container/go/cmd/puller/puller.go]: Suvanjan removed this due to importing issue, we probably shouldn't add this back now.

File: container/new_pull.bzl (hunk 1 above)
Comment (xwinxu) [reply container/new_pull.bzl]: git problem again. I switched branches but it somehow didn't push properly  :(

Comment (xwinxu) [reply container/new_pull.bzl]: I'll delete the file and try again.

File: container/go/cmd/puller/puller.go [hunk 3]
```diff
@@ -1,4 +1,4 @@
-// Copyright 2015 The Bazel Authors. All rights reserved.
+/// Copyright 2015 The Bazel Authors. All rights reserved.
```
Comment (xingao267) [inline container/go/cmd/puller/puller.go]: Try to avoid this kind of unnecessary change which is not really related to the purpose of this PR.

File: container/go/pkg/oci/BUILD [hunk 4]
```diff
@@ -1,12 +1,31 @@
...
+# limitations under the License.
+# ###
+# Build for the new writer function to write OCI Format Images to disk.
 load("@io_bazel_rules_go//go:def.bzl", "go_library")
 
 go_library(
     name = "go_default_library",
-    srcs = ["reader.go"],
+    srcs = [
+        "reader.go",
```
Comment (xingao267) [inline container/go/pkg/oci/BUILD]: Are you sure it builds? reader.go seems do not exist.

Comment (xingao267) [reply container/go/pkg/oci/BUILD]: Sorry...my bad. I missed Heddy's PR. Ignore this comment.
